config/routes.rb

```diff
@@ -1,9 +1,21 @@
 Autolab3::Application.routes.draw do
-  root "courses#index"
 
   devise_for :users, controllers: { omniauth_callbacks: "users/omniauth_callbacks" },
                      path_prefix: "auth"
 
+  devise_scope :user do
+
+    authenticated :user do
+      root 'courses#index', as: :authenticated_root
+    end
+
+    unauthenticated do
+      root 'devise/sessions#new', as: :unauthenticated_root
+    end
+
+  end
```

Comment:
It is just that the same route acts differently depending on whether the user is authenticated. We can't have them in one controller because 'devise/sessions#new' code is part of the devise gem.